library/src/main/java/com/novoda/downloadmanager/NotificationCreator.java

```diff
@@ -32,6 +32,10 @@ NotificationInformation createNotification(final T notificationPayload) {
         return new NotificationInformation() {
             @Override
             public int getId() {
+                if (notificationPayload instanceof DownloadBatchStatus) {
+                    return ((DownloadBatchStatus) notificationPayload).getDownloadBatchId().hashCode();
```

Comment:
`Object.hashcode` is returning different values each time it is executed. Adding `hashcode` to the whole `DownloadBatchStatus` lags the UI hard because it needs to calculate this for each percentage point on a large object, hence doing it on the id only.

Reply:
If you did this `.hashCode` at point of `DownloadBatchStatus` creation and you would extend that interface with an `int notificationId()`, would that solve all performance issues here and prevent the casting?

Reply:
let's think of a solution by which we do not have to calculate the hash every single time, being a known constant value, next PR